### PR TITLE
[content-service-api] add bool to snapshot intializer struct

### PR DESCRIPTION
**Pull request by sagor999**
## Description

Add a boolean to snapshot initializer so that we know when it is old style snapshot using GCS\similar or using new PVC(Volume Snapshot) approach.

## Related Issue(s)

Part of https://github.com/gitpod-io/gitpod/pull/10689

## Release Notes

```release-note
none
```

### CODE REVIEW

**Review by csweichel — APPROVED** [review]

**Comment by jankeromnes** [comment]
Hello there! 👋 Rebuild & temporary hold to unblock the merge queue:

/hold

/werft run

**Pull request merged**
